lib/streamlit/DeltaGenerator.py

```diff
@@ -99,11 +107,19 @@ def _with_element(method):
 
     @_wraps_with_cleaned_sig(method, 2)  # Remove self and element from sig.
     def wrapped_method(dg, *args, **kwargs):
+        delta_type = method.__name__
+        last_index = None
+
+        if delta_type in DELTAS_WHICH_USE_DATAFRAME and len(args) > 0:
```

Comment:
I don't think this includes the changes we talked about. This code was going to move to DeltaGenerator, and we also needed to add this to add_rows.

Reply:
Inside _enqueue_new_element_delta the data is already transformed and i'm needing the original one , that's why i did not move that .

The logics for add_rows is added